### DISCUSSION TITLE
How to wire Raspberry Pico

### DISCUSSION THREAD

**Discussion by epictechlab** [awawa-dev/HyperSerialPico#14] [General]
Hi,

I've several issues with my previous Setup Raspberry Pi 4 and esp32 running offiziell WLED Losing Wireless connection, flickering, am Hugh delay. So I've decided to try this one. But there is that thing I don't understand. Do I have to wire the spi pin together to get it worked? Which kind of wire is preferred? Is an Ethernet cable recommend because of magnet interfirence? My pi4 sits behind my TV Lowboard and the pico should Sid behind the TV.

**Comment by awawa-dev** [comment]
Hi
All you need is a USB connection from the RPi to the Pico. Pico should be placed near the LED strip (mine are approx. 50cm) and usually you do not need to worry about the interference. The connection of Pico to LEDs depends on the type of LED strip, and the pinout is described on the main project page. If you already have esp32, you can try HyperSerialESP32 or HyperSPI, although setup is not as simple as Pico/HyperSerialPico and requires a flasher. I'm not sure what flickering you're experiencing, but sometimes it can be hardware/electrical in nature: missing common ground (or ground connection is too long and has loops), data cables too long, missing proper 3.3V to 5V level shifter (do not use I2C level shifters), or these are the first signs of a faulty LED strip.

**Reply by epictechlab** [reply]
Thanks allot for that quick response. But why I shouldn't use an i2c level shifter?

**Reply by awawa-dev** [reply]
Usually they are very slow

**Comment by bouba830** [comment]
Hello , 
First of all thank you for your work and your time, but as I'm looking for the best solution I opted for a rasp pi 4b 2Go with in raspi pico rp2040. 
I ordered a scorpio for the regulator. But here's where I'm still stuck: the connections. Do you know of a page where I could find a shema? And also a little explanation of the compilation for a multi-segment with sk6812?
Thanks for your help.

Traduit avec DeepL https://www.deepl.com/app/?utm_source=android&utm_medium=app&utm_campaign=share-translation

**Comment by awawa-dev** [comment]
@bouba830 

Hi
Sorry I confused it with HyperSPI ;-) for HyperSerialPico the connection is trivial: you connect it to the HyperHDR host device via usb, and in the case of single segment and prebuilt custom firmware for Scorpio you connect 2 cables to the leds: a signal cable from pin 0 at the very end of the Scorpio board and the adjacent ground.
If you just enable multi-segment in a custom build then also pin 1 is used. For Scorpio use OUTPUT_DATA_PIN 16 and BOOT_WORKAROUND ON when compiling.

**Reply by bouba830** [reply]
Thanks for your answers :) 
So you put me in doubt lol .
I have everything I need (at least I think I do) to make the best and fastest assembly, but what is it?
It's hyper SPI in multi-segment, isn't it?
If that's the case, then I'm the one who's made a mistake and the one I'm interested in would be the SPI and not the serial port.
Dsl I bother you but I want so much to learn and realize your project.
Thank you

**Reply by awawa-dev** [reply]
Start with HyperSerialPico, it's easier :) If everything works fine with HyperSerialPico and you want to try the second solution, you can simply connect GPIOs at any time and upload HyperSPI firmware. Although HyperSPI is much faster, in most cases the difference will not be noticeable for Rp2040 (this is often different in the case of ESP32 boards & HyperSerialESP32 and HyperSPI if there is a physical UART)

**Reply by bouba830** [reply]
I understand, I'll try this quickly and get back to you to tell you how it goes.
Thank you

**Reply by bouba830** [reply]
Hello 😊,
So here I have everything together but I'm encountering some difficulties lol in sorry 😅.
I'm trying to learn but the translation from English to French is not top lol.
So I have a pi4B with a pico scorpio that I want to connect as a spinnaker on a cold sk6812 double segment.
Could you please tell me where to find the right information so that I don't make a mistake? 
Or how to do it? 
As for the SN74AHCT125N, do I need it or not? I can't find any shema with a pico.
I have a 65" screen so I have a total of 419 LEDs 😁😅.
Thanks and have a good weekend.

**Reply by awawa-dev** [reply]
Hi
> As for the SN74AHCT125N, do I need it or not? I can't find any shema with a pico.

If you have a Adafruit rp2040 scorpio then you dont need SN74AHCT125N. It has already buiilt-in level shifter.

Connection example for single segment:
https://github.com/satgit62/How-to-Install-and-set-up-Ambilight-on-LG-webOS?tab=readme-ov-file#rp2040-adalight-feather-scorpio-mit-level-shifter-ausgang-gpio-16

For multi-segment you need to compile firmware:
https://github.com/awawa-dev/HyperSerialPico/wiki

	# Use multi-segment, starting index of second led strip or OFF to disable
	set(SECOND_SEGMENT_INDEX OFF)

	# If multi-segment is used and it's reversed, set this option to ON to enable reversing
	set(SECOND_SEGMENT_REVERSED OFF)
also for Scorpio:
> OUTPUT_DATA_PIN 16 and BOOT_WORKAROUND ON

**Comment by ingbrzy** [comment]
Hello @awawa-dev  

I am building new HyperHDR + HyperSerialPico + SN74AHCT125N + SK6812 RGBW DC12V for 55" TV..

Is my wiring diagram correct?

![single_segment_pico_bb](https://github.com/user-attachments/assets/ad08585f-0972-4529-af96-9e1ead81d90d)

**Reply by awawa-dev** [reply]
If your are going to use non-genuine power supply without PD then you must use some tricks in config.txt https://github.com/raspberrypi/documentation/blob/develop/documentation/asciidoc/computers/raspberry-pi/power-supplies.adoc 

```
The Raspberry Pi 5 provides 1.6A of power to downstream USB peripherals when connected to a power supply capable of 5A at +5V (25W). When connected to any other compatible power supply, the Raspberry Pi 5 restricts downstream USB devices to 600mA of power. 
```

**Reply by ingbrzy** [reply]
> If your are going to use non-genuine power supply without PD then you must use some tricks in config.txt https://github.com/raspberrypi/documentation/blob/develop/documentation/asciidoc/computers/raspberry-pi/power-supplies.adoc
> 
> ```
> The Raspberry Pi 5 provides 1.6A of power to downstream USB peripherals when connected to a power supply capable of 5A at +5V (25W). When connected to any other compatible power supply, the Raspberry Pi 5 restricts downstream USB devices to 600mA of power. 
> ```

For RPi5 I will use original power adapter.. I just borrowed that part of the diagram from your file.. :) it is just to show different 5V and 12V power supply..
important for me is part after RPi5 if it is correct.. 
Thank you!

**Reply by awawa-dev** [reply]
Yes, it will work. If you want to reduce number of component you can also use Adafruit or Pimoroni boards with built-in level shifter.

**Reply by ingbrzy** [reply]
Thank you for your help!
I did not find clear wiring diagram for RPi Pico in you guides, so feel free to use it.. :)

fzz file
https://mega.nz/file/ux4DFIzQ#rB25NWYPqPb5BElWbKHTTcRb3IQGJrylw5QsQzuh2lw

You do a great job with HyperHDR !!!
